### PR TITLE
Feat: Allow a custom equals parameter for ObservableStream

### PR DESCRIPTION
**Pull request by amondnet**
Create a custom equals parameter for [`ObservableStream`](https://pub.dev/documentation/mobx/latest/mobx/ObservableStream-class.html), like what exists already for [`Observable`](https://pub.dev/documentation/mobx/latest/mobx/Observable/Observable.html).

Currently, `ObservableStream` uses == to determine if the value changed. This creates an option to override that with a different EqualityComparator.

https://github.com/mobxjs/mobx.dart/blob/f8677da98b2077ae22b536abe3ea11be2ba8050e/mobx/lib/src/api/async/observable_stream.dart#L439

https://github.com/mobxjs/mobx.dart/blob/f8677da98b2077ae22b536abe3ea11be2ba8050e/mobx/lib/src/core/observable.dart#L62

https://github.com/mobxjs/mobx.dart/blob/f8677da98b2077ae22b536abe3ea11be2ba8050e/mobx/lib/src/core/observable.dart#L106

https://github.com/mobxjs/mobx.dart/issues/679

### CODE REVIEW

**Review by fzyzcjy — APPROVED** [review]
LGTM

**Review by pavanpodila — CHANGES_REQUESTED** [review]
We should also add to the docs

**Review by pavanpodila — CHANGES_REQUESTED** [review]
Please add a changelog and version bump for the mobx package

**Comment by pavanpodila** [comment]
Can you resolve conflicts pls before the merge. Also will need to bump up the version for publishing and add to changelog

**Comment by amondnet** [comment]
@pavanpodila Done.

File: mobx/lib/src/api/async/observable_stream.dart [hunk 1]
```diff
@@ -332,7 +345,8 @@ class _ObservableStreamController<T> {
             name: '$name.status'),
         _valueType = Observable(_ValueType.value,
             context: context, name: '$name.valueType'),
-        _data = Observable(initialValue, context: context, name: '$name.data') {
+        _data = Observable<dynamic>(initialValue,
```
Comment (pavanpodila) [inline mobx/lib/src/api/async/observable_stream.dart:348]: why not `Observable<T?>`

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx/lib/src/api/async/observable_stream.dart (hunk 1 above)
Comment (amondnet) [reply mobx/lib/src/api/async/observable_stream.dart:348]: @pavanpodila 
https://github.com/mobxjs/mobx.dart/blob/cea7013fad1fcdfbb3213045db9be5e3f311f2f3/mobx/lib/src/api/async/observable_stream.dart#L452

Yes, I would like to use `Observable<T?>`, but `Object` type is assigned inside the `_onError` function.




```dart
  void _onError(Object error) {
    final actionInfo = _actions.startAction();
    try {
      _status.value = StreamStatus.active;
      _valueType.value = _ValueType.error;
      _data.value = error;
    } finally {
      _actions.endAction(actionInfo);
      _tryInsertInitialValue();
      _controller.addError(error);
    }
  }
```

**Review by pavanpodila — APPROVED** [review]

**Comment by amondnet** [comment]
@all-contributors please add @amondnet for doc

**Pull request merged**